source/val/validation_state.cpp

```diff
@@ -332,6 +332,24 @@ void ValidationState_t::RegisterCapability(SpvCapability cap) {
     desc->capabilities.ForEach(
         [this](SpvCapability c) { RegisterCapability(c); });
   }
+
+  switch (cap) {
+    case SpvCapabilityInt16:
+      features_.declare_int16_type = true;
+      break;
+    case SpvCapabilityFloat16:
+    case SpvCapabilityFloat16Buffer:
+      features_.declare_float16_type = true;
+      break;
+    case SpvCapabilityStorageUniformBufferBlock16:
+    case SpvCapabilityStorageUniform16:
+    case SpvCapabilityStoragePushConstant16:
+    case SpvCapabilityStorageInputOutput16:
+      features_.declare_int16_type = true;
```

Comment:
Is this being too permissive? From my understanding, `SpvCapabilityInt16` enables more operations on int16 than `SpvCapabilityStorage*16`, but we don't actually distinguish that here. But we can handle this later maybe.